scripts/monorepo/src/getDependencies.js

```diff
@@ -123,16 +126,21 @@ function collectDependencies(
       return;
     }
 
-    _acc.push({
+    collectedDeps.push({
       name: dependency.target,
       dependencyType: dependency.dependencyType,
       isTopLevel: _areTopLevelDeps,
     });
+  });
+
+  // update main dep stack with actual collected deps
+  _acc.push(...collectedDeps);
 
-    if (!options.shallow) {
```

Comment:
whenever the same dependency deeper within the tree was present it overrode parent tree - giving us wrong result in terms of `isTopLevel` metadata.